galley/pkg/source/kube/schema/resourcespec.go

```diff
@@ -51,6 +48,36 @@ type ResourceSpec struct {
 
 	// The converter to use
 	Converter converter.Fn
+
+	// Versions of the K8s resource
+	Versions []CrdVersion
+
+	// Version of the K8s resource
+	Version string
+}
+
+type CrdVersion struct {
+	// Name is the version name, e.g. “v1”, “v2beta1”, etc.
+	Name string
+
+	// Served is a flag enabling/disabling this version from being served via REST APIs
+	Served bool
```

Comment:
Why do we need served or storage? Where will they be used in the code?

Reply:
We need this because for multiple version CRD, we can have one and only one storage version but multiple served version.  I will use this flag to query the version when we generating API resource in galley.